### DISCUSSION TITLE
Separation of Deployment from Configuration in Fybrik

### DISCUSSION THREAD

**Discussion by cdoron** [fybrik/fybrik#943] [Developers]
Currently, there is no way to update configuration of FybrikModules (e.g. arrow-flight-module) without re-deploying helm-charts.

This causes several problems:
1. Deploying helm chart is costly in term of performance.
2. Some services must not be re-deployed while running due to security concerns
3. There may be a delay until a new configuration takes effect (for instance, if configuration is done through a mounted configuration map)

**Our objective is to support configuration updates without re-deployment of helm-charts.** This could be done, for instance, by running configuration servers from which fybrik modules would get their configuration (through pull or push notifications).

Two phases to reach the objective:
1. Restructure 'values' sent to helm during deployment (more on this later).
2. Choose and implement configuration-update mechanism

Regarding phase 1: when the blueprint manager performs a "helm install", it sends a set of values to helm. These values are taken from the fybrik-module CRD (`chart.values` field) and taken from other sources such as the data catalog. The first phase would be to separate 'deployment values' from 'configuration values' in the 'values' sent to helm during deployment.

**Comment by revit13** [comment]
if a dataset is been removed from the values sent to the module how it should be done? currently, it will not appear in its values. The answer could be trivial just wanted to know it :-) Thanks

**Reply by shlomitk1** [reply]
@revit13 The idea is to separate dataset info and additional values that can be specified in FybrikModule spec in chart.values. Both are going to the module but may be something like .Values.deployment.xxx and .Values.config.yyy
